base/utils/R/ensemble.R

```diff
@@ -130,15 +134,32 @@ get.ensemble.samples <- function(ensemble.size, pft.samples, env.samples,
                                total.sample.num)
     }
     
+    
     ensemble.samples <- list()
     
+    
     col.i <- 0
     for (pft.i in seq(pft.samples)) {
       ensemble.samples[[pft.i]] <- matrix(nrow = ensemble.size, ncol = length(pft.samples[[pft.i]]))
+      
+      # meaning we want to keep MCMC samples together
+      if(length(pft.samples[[pft.i]])>0 & !is.null(param.names)){ 
+        # TODO: for now we are sampling row numbers uniformly
+        # throw a warning if other methods were requested 
+        if(method != "uniform"){
+          PEcAn.logger::logger.warn("Other approaches are not implemented yet for sampling jointly. Defaulting to uniform sampling of the chain.")
```

Comment:
Probably want this to be a logger.severe -- if downstream code is expecting a different sampling method (e.g. sobol SA) then returning random could cause a but that's hard to decode.